test/framework/easyconfig.py

```diff
@@ -1204,6 +1204,32 @@ def test_dump_extra(self):
 
         dumped_ec = EasyConfig(testec)
 
+    def test_dump_template(self):
+        """ Test EasyConfig's dump() method for files containing templates"""
+        rawtxt = '\n'.join([
+            'easyblock = "EB_toy"',
+            '',
+            'name = "foo"',
+            'version = "0.0.1"',
+            '',
+            'homepage = "http://foo.com/"',
+            'description = "foo description"',
+            '',
+            'toolchain = {\'version\': \'dummy\', \'name\': \'dummy\'}',
+            '',
+            'sources = [\'%(namelower)s-%(version)s.TAR.gz\']',
```

Comment:
what happens if you make this:

```python
sources = ['foo-0.0.1.tar.gz']
```

In a first pass, this should become `'%(name)s-%(version)s.tar.gz'`, which could then be translated to `SOURCE_TAR_GZ`...

That would mean doing multiple passes with `replace_templates` though, until there are no more changes...

Is that feasible?